jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/HttpInput.java

```diff
@@ -311,9 +312,7 @@ private int read(ByteBuffer buffer, byte[] b, int off, int len) throws IOExcepti
                 LOG.debug("read error={} {}", error, this);
             if (error != null)
             {
-                if (error instanceof IOException)
-                    throw (IOException)error;
-                throw new IOException(error);
+                ExceptionUtil.ifExceptionThrowAs(IOException.class, error);
```

Comment:
Why not `throw IO.rethrow(error)`?
It shows better the control flow and it's less verbose.

Reply:
That changes the logic a little bit... which I didn't want to do in a documentation PR. This was just a cleanup.   But it is probably a good change. I'll make it and see if anything breaks.

Reply:
Actually, I'll keep it like this for EE9, but will rethrow in EE10